pandas/tests/reshape/concat/test_categorical.py

```diff
@@ -203,6 +206,33 @@ def test_categorical_concat_gh7864(self):
         dfa = df1._append(df2)
         tm.assert_index_equal(df["grade"].cat.categories, dfa["grade"].cat.categories)
 
+    @pytest.mark.parametrize(
+        ("input_1", "input_2", "expected"),
+        (  # both Catergorical dtypes with datetimes
+            (
+                DataFrame(
+                    {"x": Series(datetime(2021, 1, 1), index=[0], dtype="category")}
+                ),
+                DataFrame(
+                    {"x": Series(datetime(2021, 1, 2), index=[1], dtype="category")}
+                ),
+                DataFrame({"x": Series([datetime(2021, 1, 1), datetime(2021, 1, 2)])}),
+            ),
+            # both Catergorical dtypes, one datetime, one string
+            (
+                DataFrame(
+                    {"x": Series(datetime(2021, 1, 1), index=[0], dtype="category")}
+                ),
+                DataFrame({"x": Series("test string", index=[1], dtype="category")}),
+                DataFrame({"x": Series([datetime(2021, 1, 1), "test string"])}),
+            ),
+        ),
+    )
+    def test_categorical_datetime_concat(self, input_1, input_2, expected):
```

Comment:
Please write the cases down explicitly. We do not gain anything through the parametrization and this is really hard to read

Reply:
I can make those changes. Before setting off the CI to fail again though, as mentioned in [this comment](https://github.com/pandas-dev/pandas/issues/39443#issuecomment-823964197), my PyTests are passing locally, but not on the CI. Could do with some help on why this might be? 

Locally I get below to pass:

```python
    def test_categorical_datetime_concat(self):
        # GH 39443
        # test catergorical dataframes both containing datetimes
        df1 = DataFrame(
            {"x": Series(datetime(2021, 1, 1), index=[0], dtype="category")}
        )
        df2 = DataFrame(
            {"x": Series(datetime(2021, 1, 2), index=[1], dtype="category")}
        )
        expected = DataFrame(
            {"x": Series([datetime(2021, 1, 1), datetime(2021, 1, 2)])}
        )
        result = pd.concat([df1, df2])
        tm.assert_equal(result, expected)
```

But in the CI (based on previous failure) - this will fail with: 

```
E       AssertionError: Attributes of DataFrame.iloc[:, 0] (column name="x") are different
E       
E       Attribute "dtype" are different
E       [left]:  CategoricalDtype(categories=['2021-01-01', '2021-01-02'], ordered=False)
E       [right]: datetime64[ns]
```

Looks like it links to #14016